deepchem/models/torch_models/gcn.py

```diff
@@ -0,0 +1,350 @@
+                                     batchnorm=[batchnorm] * num_gnn_layers,
+                                     dropout=[dropout] * num_gnn_layers,
+                                     n_tasks=out_size,
+                                     predictor_hidden_feats=predictor_hidden_feats,
+                                     predictor_dropout=predictor_dropout)
+
+    def forward(self, g):
+        """Predict graph labels
+
+        Parameters
+        ----------
+        g: DGLGraph
+            A DGLGraph for a batch of graphs. It stores the node features in
+            ``dgl_graph.ndata[self.nfeat_name]``.
+
+        Returns
+        -------
+        torch.Tensor
+            The model output.
+
+            * When self.mode = 'regression',
+              its shape will be ``(dgl_graph.batch_size, self.n_tasks)``.
+            * When self.mode = 'classification', the output consists of probabilities
+              for classes. Its shape will be
+              ``(dgl_graph.batch_size, self.n_tasks, self.n_classes)`` if self.n_tasks > 1;
+              its shape will be ``(dgl_graph.batch_size, self.n_classes)`` if self.n_tasks is 1.
+        torch.Tensor, optional
+            This is only returned when self.mode = 'classification', the output consists of the
+            logits for classes before softmax.
+        """
+        node_feats = g.ndata[self.nfeat_name]
+        out = self.model(g, node_feats)
+
+        if self.mode == 'classification':
+            if self.n_tasks == 1:
+                logits = out.view(-1, self.n_classes)
+                softmax_dim = 1
+            else:
+                logits = out.view(-1, self.n_tasks, self.n_classes)
+                softmax_dim = 2
+            proba = F.softmax(logits, dim=softmax_dim)
+            return proba, logits
+        else:
+            return out
+
+class GCNModel(TorchModel):
+    """Model for Graph Property Prediction Based on Graph Convolution Networks (GCN).
+
+    This model proceeds as follows:
+
+    * Update node representations in graphs with a variant of GCN
+    * For each graph, compute its representation by 1) a weighted sum of the node
+      representations in the graph, where the weights are computed by applying a
+      gating function to the node representations 2) a max pooling of the node
+      representations 3) concatenating the output of 1) and 2)
+    * Perform the final prediction using an MLP
+
+    Examples
+    --------
+
+    >>> import deepchem as dc
+    >>> from deepchem.models import GCNModel
+    >>> featurizer = dc.feat.MolGraphConvFeaturizer()
+    >>> tasks, datasets, transformers = dc.molnet.load_tox21(
+    ...     reload=False, featurizer=featurizer, transformers=[])
+    >>> train, valid, test = datasets
+    >>> model = dc.models.GCNModel(mode='classification', n_tasks=len(tasks),
+    ...                            number_atom_features=30, batch_size=32, learning_rate=0.001)
+    >>> model.fit(train, nb_epoch=50)
```

Comment:
Can you change the example like this?
We tests all examples written in docstring. But `load_tox21 ` is slow, so we should skip this example.

```
>>>
>> import deepchem as dc
>> from deepchem.models import GCNModel
>> featurizer = dc.feat.MolGraphConvFeaturizer()
>> tasks, datasets, transformers = dc.molnet.load_tox21(
..     reload=False, featurizer=featurizer, transformers=[])
>> train, valid, test = datasets
>> model = dc.models.GCNModel(mode='classification', n_tasks=len(tasks),
 ..                            number_atom_features=30, batch_size=32, learning_rate=0.001)
>> model.fit(train, nb_epoch=50)
```

This style renders the example correctly in docs and skips the doctest.

Reply:
Done.